src/Nethermind/Nethermind.Monitoring/Metrics/MetricsController.cs

```diff
@@ -16,41 +16,79 @@
     {
+        public static readonly Dictionary<string, string> StaticTags= new();
+
         private readonly int _intervalSeconds;
         private Timer _timer;
-        private Dictionary<string, Gauge> _gauges = new Dictionary<string, Gauge>();
-        private Dictionary<Type, (PropertyInfo, string)[]> _propertiesCache = new Dictionary<Type, (PropertyInfo, string)[]>();
-        private Dictionary<Type, (FieldInfo, string)[]> _fieldsCache = new Dictionary<Type, (FieldInfo, string)[]>();
-        private Dictionary<Type, (string DictName, IDictionary<string, long> Dict)> _dynamicPropCache = new Dictionary<Type, (string DictName, IDictionary<string, long> Dict)>();
-        private HashSet<Type> _metricTypes = new HashSet<Type>();
+        private Dictionary<string, Gauge> _gauges = new();
+        private Dictionary<Type, (PropertyInfo, string)[]> _propertiesCache = new();
+        private Dictionary<Type, (FieldInfo, string)[]> _fieldsCache = new();
+        private Dictionary<Type, (string DictName, IDictionary<string, long> Dict)> _dynamicPropCache = new();
+        private HashSet<Type> _metricTypes = new();
 
         public void RegisterMetrics(Type type)
         {
             EnsurePropertiesCached(type);
             foreach ((PropertyInfo propertyInfo, string gaugeName) in _propertiesCache[type])
             {
-                _gauges[gaugeName] = CreateGauge(BuildGaugeName(propertyInfo.Name));
+                GetMemberInfoMectricsDescription(propertyInfo, gaugeName);
             }
 
             foreach ((FieldInfo fieldInfo, string gaugeName) in _fieldsCache[type])
             {
-                _gauges[gaugeName] = CreateGauge(BuildGaugeName(fieldInfo.Name));
+                GetMemberInfoMectricsDescription(fieldInfo, gaugeName);
             }
 
             _metricTypes.Add(type);
         }
 
+        private void GetMemberInfoMectricsDescription(MemberInfo propertyInfo, string gaugeName)
+        {
+            GaugeConfiguration configuration = new();
+            Dictionary<string, string> tagValues = new();
+
+            propertyInfo.GetCustomAttributes<MetricsStaticDescriptionTagAttribute>()
+                .ForEach(attribute => InsertStaticMemberInfo(attribute.Informer, attribute.Label, tagValues));
+            configuration.StaticLabels = tagValues;
+
+            string description = propertyInfo.GetCustomAttribute<DescriptionAttribute>()?.Description;
+            _gauges[gaugeName] = CreateGauge(BuildGaugeName(propertyInfo.Name), description, configuration);
+        }
+
+        private static void InsertStaticMemberInfo(Type givenInformer, string givenName, Dictionary<string, string> tagValues)
+        {
+            Type type = givenInformer;
+            PropertyInfo[] tagsData = type.GetProperties(BindingFlags.Static | BindingFlags.Public);
+            PropertyInfo info = tagsData.FirstOrDefault(info => info.Name == givenName);
+            if (info == null)
```

Comment:
recently we went with standarizing `is` for null comparisions

Reply:
Ok

Reply:
Fixed all over the solution.